lib/cequel/metal/data_set.rb

```diff
@@ -655,12 +673,16 @@ def row_specifications_cql
 
       protected
 
-      attr_writer :row_limit, :query_consistency
+      attr_writer :row_limit, :query_consistency, :query_page_size, :query_paging_state
 
       private
 
       def execute_cql(cql, *bind_vars)
-        keyspace.execute_with_consistency(cql, bind_vars, query_consistency)
+        keyspace.execute_with_options(cql, bind_vars, {
+          consistency: query_consistency, 
+          page_size: query_page_size, 
+          paging_state: query_paging_state
+        })
```

Comment:
Hmm... I think i see your point. I would be helpful, for me, to see some examples of expected usage pattern. That would clarify the user experience this would create. A test or two might do the trick.